### ISSUE TITLE
0.4.1: panic on receiving empty Close frame

### ISSUE THREAD

**Issue by audunhalland** [jgraef/reqwest-websocket#24]
I'm using an axum websocket server, calling `axum::extract::ws::WebSocket::close()`. This is a close with a `None` CloseFrame.

the client, `reqwest-websocket` tries to convert the underlying `tungstenite` message into reqwest-websocket's own `Message` type. This happens first in `impl Stream for WebSocket` (`lib.rs`), which calls `TryFrom<tungstenite::Message> for Message` (`native.rs`).

The code there is:
```
            tungstenite::Message::Close(None) | tungstenite::Message::Frame(_) => {
                Err(FromTungsteniteMessageError { original: value })
            }
```

This `Err` is turned into a panic in the `Stream` implementation:

```
                    Err(e) => {
                        // this fails only for raw frames (which are not received)
                        panic!("Received an invalid frame: {e}");
                   }
```

**Comment by jgraef** [comment]
Thanks for the bug report!

I think there's no good fix without a breaking change, because our `Message::Close` expects a close code and reason. We can either:

- Catch the specific error in `poll_next` and return `None`.
- Return a default `Message::Close { code: CloseCode::default(), reason: "".to_owned() }`.

I think I'll do the latter. I'll push a fix in a few minutes.

**Comment by jgraef** [comment]
fyi, the patch for this is now released in version 0.4.2
